api/osrd_infra/views/pathfinding.py

```diff
@@ -146,45 +146,77 @@ def parse_steps_input(steps):
             if begin < end:
                 for interval in sorted(tree.overlap(begin, end)):
-                    add_vmax_result(result, offset, interval.data)
+                    add_chart_point(result, offset, interval.data, field_name)
                     offset += abs(max(begin, interval.begin) - min(end, interval.end))
-                    add_vmax_result(result, offset, interval.data)
+                    add_chart_point(result, offset, interval.data, field_name)
             else:
                 for interval in reversed(sorted(tree.overlap(end, begin))):
-                    add_vmax_result(result, offset, interval.data)
+                    add_chart_point(result, offset, interval.data, field_name)
                     offset += abs(max(end, interval.begin) - min(begin, interval.end))
-                    add_vmax_result(result, offset, interval.data)
+                    add_chart_point(result, offset, interval.data, field_name)
     return result
 
 
+def compute_vmax(payload, track_map):
+    tree_vmax = defaultdict(IntervalTree)
+    for track_id, track in track_map.items():
+        # Set a default vmax of -1 (need to be override by the rolling stock max speed)
+        tree_vmax[track_id].addi(0, track.track_section.length, -1)
+        for range_component in track.range_objects.all():
+            start = range_component.start_offset
+            end = range_component.end_offset
+            speed = range_component.entity.speed_section_part.speed_section.speed_section.speed
+            tree_vmax[track.entity_id].chop(start, end)
+            tree_vmax[track.entity_id].addi(start, end, speed)
+    return create_chart(payload["path"], tree_vmax, "speed")
+
+
+def compute_slopes(payload, track_map):
+    tree_slopes = defaultdict(IntervalTree)
+    for track_id, track in track_map.items():
+        # Set a default gradient of 0
+        tree_slopes[track_id].addi(0, track.track_section.length, 0)
+        for slope_component in track.slope_set.all():
+            start = slope_component.start_offset
+            end = slope_component.end_offset
+            gradient = slope_component.gradient
+            tree_slopes[track.entity_id].chop(start, end)
+            tree_slopes[track.entity_id].addi(start, end, gradient)
+    return create_chart(payload["path"], tree_slopes, "gradient")
+
+
+def compute_curves(payload, track_map):
+    tree_curves = defaultdict(IntervalTree)
+    for track_id, track in track_map.items():
+        # Set a default radius of 0
+        tree_curves[track_id].addi(0, track.track_section.length, 0)
+        for curve_component in track.curve_set.all():
+            start = curve_component.start_offset
+            end = curve_component.end_offset
+            radius = curve_component.radius
+            tree_curves[track.entity_id].chop(start, end)
+            tree_curves[track.entity_id].addi(start, end, radius)
+    return create_chart(payload["path"], tree_curves, "radius")
```

Comment:
I feel like we could factorize more than this, there aren't a lot of variations in those functions